src/DGSM.jl

```diff
@@ -39,11 +48,44 @@ function DGSM(f,samples::Int,distr::AbstractArray)
     
     
     #Evaluating E(a) E(|a|) and E(a^2)
+
+    a = [mean(dfdx[:,i]) for i in 1:k]
+    asq = [mean(dfdx[:,i].^2) for i in 1:k]
+    absa = [mean(abs.(dfdx[:,i])) for i in 1:k]
+    
     
-    DGSM_Vi = [DGSM(mean(dfdx[:,i]),mean(abs.(dfdx[:,i])),mean(dfdx[:,i].^2)) for i in 1:k]
     
-    #This function finally returns an array of structures, consisting a, absa and asq
-    #respectively for the k independent parameters
+    if crossed == true
+        
+        #Evaluating the derivatives with AD
+        dfdxdy = [hess(XX[i]) for i in 1 : samples]
     
+        crossed = zeros(Float64,k,k)
+        crossedsq = zeros(Float64,k,k)
+        abscrossed = zeros(Float64,k,k)
+    
+        #computing the elements of crossed, crossedsq, abscrossed
+    
+        for a in 1:k
+            for b in a+1:k
+                crossed[b + (a-1)*k] = mean(dfdxdy[i][b + (a-1)*k] for i in 1:samples)
+                crossed[a + (b-1)*k] = crossed[b + (a-1)*k]
+                crossedsq[b + (a-1)*k] = mean(dfdxdy[i][b + (a-1)*k]^2 for i in 1:samples)
+                crossedsq[a + (b-1)*k] = crossedsq[b + (a-1)*k]
+                abscrossed[b + (a-1)*k] = mean(abs(dfdxdy[i][b + (a-1)*k]) for i in 1:samples)
+                abscrossed[a + (b-1)*k] = abscrossed[b + (a-1)*k]
+            end
+        end
+        
+    else
+        crossed = zeros(Float64,0,0)
+        crossedsq = zeros(Float64,0,0)
+        abscrossed = zeros(Float64,0,0)
```

Comment:
Instead just use `DGSM(a, absa, asq, nothing, nothing, nothing)` and fix the type constraint in the struct as well..